### PR TITLE
Http3 nits

### PR DESCRIPTION
**Pull request by ddragana**
Http3 nits:
- move API functions to single place,
- make some functions easier to read,
- rename read_data and get_headers into read_response_data, get_reasponse_headers (the names are more descriptive)

### CODE REVIEW

File: neqo-client/src/main.rs [hunk 1]
```diff
@@ -202,7 +202,7 @@ impl Handler for PostConnectHandler {
                         return false;
                     }
 
-                    let headers = client.get_headers(stream_id);
+                    let headers = client.get_response_headers(stream_id);
```
Comment (martinthomson) [inline neqo-client/src/main.rs]: is this `read_response_headers()` or an accessor?

File: neqo-http3/src/connection.rs [hunk 2]
```diff
@@ -187,6 +187,19 @@ impl ::std::fmt::Display for Http3Connection {
     }
 }
 
+macro_rules! transaction {
+    ($t:expr) => {
+        match $t {
+            Some(t) => t,
+            _ => {
+                return Err(Error::TransportError(
+                    neqo_transport::Error::InvalidStreamId,
```
Comment (martinthomson) [inline neqo-http3/src/connection.rs]: If this code is generating the error, does it really make sense to use a TransportError?

File: neqo-client/src/main.rs (hunk 1 above)
Comment (ddragana) [reply neqo-client/src/main.rs]: it should be read here. But I could actually make it an accessor. I will do that in a separate issue.

File: neqo-http3/src/connection.rs (hunk 2 above)
Comment (ddragana) [reply neqo-http3/src/connection.rs]: having a transaction should match having a quic stream, therefore I use this error. I could make a new one.

File: neqo-client/src/main.rs (hunk 1 above)
Comment (ddragana) [reply neqo-client/src/main.rs]: I will do this in a separate PR, because I think I will refactor transaction_client/server) a bit more.

**Comment by agrover** [comment]
@ddragana is this PR still active or has it been superseded?

**Comment by ddragana** [comment]
I will close this PR. I needed to move forward with other that touch the same code, so this one now needs a lot of rebasing. I will just make a new one

**Pull request closed without merging**